src/main/java/com/blamejared/crafttweaker/impl/world/MCWorld.java

```diff
@@ -0,0 +1,149 @@
+    public long getTime() {
+        return internal.getGameTime();
+    }
+
+    @ZenCodeType.Getter("seaLevel")
+    public int getSeaLevel() {
+        return internal.getSeaLevel();
+    }
+
+    @ZenCodeType.Getter("raining")
+    public boolean isRaining() {
+        return internal.isRaining();
+    }
+
+    @ZenCodeType.Getter("thundering")
+    public boolean isThundering() {
+        return internal.isThundering();
+    }
+
+    @ZenCodeType.Getter("hardcore")
+    public boolean isHardcore() {
+        return internal.getWorldInfo().isHardcore();
+    }
+
+    @ZenCodeType.Getter("difficulty")
+    public String getDifficulty() {
+        return internal.getWorldInfo().getDifficulty().getTranslationKey();
+    }
+
+    @ZenCodeType.Getter("difficultyLocked")
+    public boolean isDifficultyLocked() {
+        return internal.getWorldInfo().isDifficultyLocked();
+    }
+
+    @ZenCodeType.Getter("dimension")
+    public String getDimension() {
+        return internal.getDimensionKey().getRegistryName().toString();
+    }
+
+    @ZenCodeType.Method
+    public boolean isRainingAt(MCBlockPos pos) {
+        return internal.isRainingAt(pos.getInternal());
+    }
+
+    @ZenCodeType.Method
+    public int getStrongPower(MCBlockPos pos) {
+        return internal.getStrongPower(pos.getInternal());
+    }
+
+    @ZenCodeType.Method
+    public int getRedstonePower(MCBlockPos pos, MCDirection direction) {
+        return internal.getRedstonePower(pos.getInternal(), direction.getInternal());
+    }
+
+    @ZenCodeType.Method
+    public int getRedstonePowerFromNeighbors(MCBlockPos pos) {
+        return internal.getRedstonePowerFromNeighbors(pos.getInternal());
+    }
+
+    @ZenCodeType.Method
+    public IData getTileData(MCBlockPos pos) {
+        CompoundNBT nbt = new CompoundNBT();
+        TileEntity te = internal.getTileEntity(pos.getInternal());
+        return te == null ? new MapData() : NBTConverter.convert(te.write(nbt));
+    }
+
+    @ZenCodeType.Method
+    public boolean setBlockState(MCBlockPos pos, MCBlockState state) {
+        return internal.setBlockState(pos.getInternal(), state.getInternal());
+    }
+
+    @ZenCodeType.Method
+    public MCBlockState getBlockState(MCBlockPos pos) {
+        return new MCBlockState(internal.getBlockState(pos.getInternal()));
+    }
+
+    @ZenCodeType.Method
+    public boolean isBlockPowered(MCBlockPos pos) {
+        return internal.isBlockPowered(pos.getInternal());
+    }
+
+    @ZenCodeType.Method
+    public MCBiome getBiome(MCBlockPos pos) {
+        return new MCBiome(internal.getBiome(pos.getInternal()));
+    }
+
+    @ZenCodeType.Method
+    public int nextRandomInt() {
+        return internal.rand.nextInt();
+    }
+
+    @ZenCodeType.Method
+    public int nextRandomInt(int bound) {
+        return internal.rand.nextInt(bound);
+    }
+
+    @ZenCodeType.Method
+    public boolean nextRandomBoolean() {
+        return internal.rand.nextBoolean();
+    }
```

Comment:
Please remove these random methods.

Random needs to be talked a lot more about, I'm super happy with the rest of the PR, but these need way more discussion that will take time